code/game/machinery/doors/airlock.dm

```diff
@@ -500,6 +501,81 @@ About the new airlock wires panel:
 				sleep(6)
 				update_icon(AIRLOCK_CLOSED)
 
+
+/obj/machinery/door/airlock/proc/change_paintjob(obj/item/airlock_painter/W, mob/user)
+	if(!paintable)
+		to_chat(user, "<span class='warning'>You cannot paint this type of airlock.</span>")
+		return FALSE
+
+	var/list/optionlist
+	if(airlock_material == "glass")
+		optionlist = list("Standard", "Public", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Science", "Mining", "Maintenance", "External", "External Maintenance")
+	else
+		optionlist = list("Standard", "Public", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Freezer", "Science", "Mining", "Maintenance", "External", "External Maintenance")
+
+	var/paintjob = input(user, "Please select a paintjob for this airlock.") in sortList(optionlist)
+	if((!in_range(src, usr) && loc != usr) || !W.paint(user))
```

Comment:
might be better to use user instead of usr, since user is already defined and you can stay consistent

Reply:
makes sense